MdeModulePkg/Universal/Console/ConSplitterDxe/ConSplitter.c

```diff
@@ -4873,10 +4878,13 @@ ConSplitterTextOutSetMode (
   //
   TextOutModeMap = Private->TextOutModeMap + Private->TextOutListCount * ModeNumber;
   for (Index = 0, ReturnStatus = EFI_SUCCESS; Index < Private->CurrentNumberOfConsoles; Index++) {
-    Status = Private->TextOutList[Index].TextOut->SetMode (
+    // MU_CHANGE: only skip if the flag is true and current mode matches the required one
+    if ((mPerformingDeviceAdd == FALSE) || (TextOutModeMap[Index] != Private->TextOutList[Index].TextOut->Mode->Mode)) {
+      Status = Private->TextOutList[Index].TextOut->SetMode (
                                                     Private->TextOutList[Index].TextOut,
                                                     TextOutModeMap[Index]
                                                     );
+    }
     if (EFI_ERROR (Status)) {

```

Comment:
Thanks to the "if" this has the potential to no longer be initialized.

Reply:
Addressed now. Thanks.